include/deal.II/fe/mapping_fe.h

```diff
@@ -0,0 +1,741 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2001 - 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef __deal2__mapping_fe_h
+#define __deal2__mapping_fe_h
+
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/table.h>
+#include <deal.II/fe/mapping.h>
+#include <deal.II/fe/fe.h>
+#include <deal.II/base/qprojector.h>
+#include <deal.II/base/thread_management.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+/*!@addtogroup mapping */
+/*@{*/
+
+/**
+ * The MappingFE is a generalization of the MappingQEulerian class, for arbitrary
+ * vectorial finite elements. The main difference is that this class uses a vector
+ * of absolute positions instead of a vector of displacement.
+ * In particular we think of a collections of a FE_Q or
+ * Bezier finite element (FE_Bernstein) repeated a number of times equal to the space
+ * dimension. The idea is to construct the mapping using a vector of control
+ * points, a DoFHandler associated to the geometry of the problem and a
+ * ComponentMask that tells us which components to use for the mapping.
+ * This mapping will grab from the DoFHandler the finite element, or better
+ * the collection of finite elements, to compute the mapping shape functions.
+ * So we will have two different Finite Element and DoFHandler, one for the
+ * solution field and one to describe the geometry of the problem. Historically
+ * in the deal.II library there was not such a distinction. The differences
+ * between this mapping and the MappingQ class are quite important.
+ * The MappingFE, being a generalization, requires a higher level of abstraction.
+ * This is the reason why it takes a DoFHandler and a vector of control points
+ * that are the coefficients of the shape function (so in general it is a vector
+ * of coefficient).
+ *
+ *
+ * Typically, the DoFHandler operates on a finite element that is constructed
+ * as a system element (FESystem) from continuous FE_Q() objects. An example
+ * is shown below:
+ * @code
+ *    const FE_Q<dim,spacedim> feq(1);
+ *    const FESystem<dim,spacedim> fesystem(feq, spacedim);
+ *    DoFHandler<dim,spacedim> dhq(triangulation);
+ *    dhq.distribute_dofs(fesystem);
+ *    Vector<double> eulerq(dhq.n_dofs());
+ *    const ComponentMask mask(spacedim, true);
+ *    MappingFE<dim,spacedim> map(eulerq, dhq, mask);
+ *    map.update_euler_vector_using_triangulation(eulerq);
```

Comment:
The first of these lines are all clear but can you add a sentence that explains what the last line does?